src/hpp/corbaserver/affordance/affordance.py

```diff
@@ -40,8 +44,8 @@ def __init__(self):
 # analysis tools more intuitive for the user.
 class AffordanceTool(object):
     # Constructor
-    def __init__(self):
-        self.client = CorbaClient()
+    def __init__(self, context=None):
+        self.client = CorbaClient(context)
```

Comment:
As a minor remark, it might be more interesting to do
```python
    def __init__(self, client=None):
        self.client = CorbaClient() if client is None else client
```
so that you can pass the same instance of `CorbaClient` to several objects.